src/mem3_shards.erl

```diff
@@ -723,5 +821,63 @@ wait_writer_result(WRef) ->
 spawn_link_mock_writer(Db, Shards, Timeout) ->
     erlang:spawn_link(fun() -> shard_writer(Db, Shards, Timeout) end).
 
+keys_to_remove(Keys, N) ->
+    keys_to_remove(Keys, N, []).
+
+keys_to_remove(_Keys, 0, Acc) ->
+    Acc;
+keys_to_remove([], _N, Acc) ->
+    Acc;
+keys_to_remove(Keys, N, Acc) ->
+    Key = oneof(Keys),
+    keys_to_remove(Keys -- [Key], N - 1, [Key | Acc]).
+
+oneof(Options) ->
+    lists:nth(random:uniform(length(Options)), Options).
+
+remove_from_cache(Pid, DbName) ->
+    gen_server:cast(Pid, {cache_remove, DbName}).
+
+wait_ets_table(Pid) ->
+    wait_ets_table(Pid, _Timeout = 5000, _Delay = 50).
+
+wait_ets_table(Pid, Timeout, Delay) ->
+    case sys:get_state(Pid) of
+        {st, _, _, _, _, _, {make_public, _Ref}} ->
+            ok = timer:sleep(Delay),
+            wait_ets_table(Pid, Timeout, Delay);
+        {st, _, _, _, _, _, nil} ->
+            ok;
+        _St ->
+            ok = timer:sleep(Delay),
+            wait_ets_table(Pid, Timeout, Delay)
+    end.
```

Comment:
It seems like `Timeout` is not actually doing anything here. I would expect to see both recursive calls to look something like `wait_ets_table(Pid, Timeout - Delay, Delay),` and for there to be an additional function clause that looks like
```
wait_ets_table(_Pid, Timeout, _Delay) when Timeout < 0 ->
    throw(wait_ets_table_timeout);
```